File: official/docs/go/pickups/create_pickup.go
package example

import (
	"fmt"
	"os"
	"time"

	"github.com/EasyPost/easypost-go/v2"
)

func main() {
	apiKey := os.Getenv("EASYPOST_API_KEY")
	client := easypost.New(apiKey)

	address, err := client.GetAddress("adr_...")
	shipment, err := client.GetShipment("shp_...")
	minPickupTime := time.Now()                         // now
	maxPickupTime := time.Now().Add(time.Hour * 24 * 7) // 7 days from now

	pickup, err := client.CreatePickup(
		&easypost.Pickup{
			IsAccountAddress: false,
			Address:          address,
			Shipment:         shipment,
			MinDatetime:      &minPickupTime,
			MaxDatetime:      &maxPickupTime,
			Instructions:     "Special pickup instructions",
		},
	)

	fmt.Println(pickup)
}

File: official/docs/csharp/current/pickups/create-pickup.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

            var client = EasyPost.Client(apiKey);

            Pickup pickup = await client.Pickup.Create(new Dictionary<string, object>()
            {
                { "reference", "my-first-pickup" },
                { "min_datetime", "2022-10-01 10:30:00" },
                { "max_datetime", "2022-10-01 17:30:00" },
                { "shipment", "shp_..." },
                { "address", "adr_..." },
                { "is_account_address", false },
                { "instructions", "Special pickup instructions" },
            });

            Console.WriteLine(JsonConvert.SerializeObject(pickup, Formatting.Indented));
        }
    }
}
